e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/check-details.spec.js

```diff
@@ -0,0 +1,169 @@
+import effectiveFromDatePage from '../../pages/deal-cancellation/effective-from-date';
+import dateConstants from '../../../../../e2e-fixtures/dateConstants';
+import bankRequestDatePage from '../../pages/deal-cancellation/bank-request-date';
+import checkDetailsPage from '../../pages/deal-cancellation/check-details';
+import reasonForCancellingPage from '../../pages/deal-cancellation/reason-for-cancelling';
+
+context('Deal cancellation - check details', () => {
+  let dealId;
+  const dealFacilities = [];
+
+  before(() => {
+    cy.insertOneDeal(MOCK_DEAL_AIN, BANK1_MAKER1).then((insertedDeal) => {
+      dealId = insertedDeal._id;
+
+      const { dealType, mockFacilities } = MOCK_DEAL_AIN;
+
+      cy.createFacilities(dealId, [mockFacilities[0]], BANK1_MAKER1).then((createdFacilities) => {
+        dealFacilities.push(...createdFacilities);
+      });
+
+      cy.submitDeal(dealId, dealType, PIM_USER_1);
+    });
+  });
+
+  after(() => {
+    cy.deleteDeals(dealId, ADMIN);
+    dealFacilities.forEach((facility) => {
+      cy.deleteFacility(facility._id, BANK1_MAKER1);
+    });
+  });
+
+  describe('when logged in as a PIM user', () => {
+    beforeEach(() => {
+      cy.login(PIM_USER_1);
+      cy.visit(relative(`/case/${dealId}/deal`));
+
+      caseDealPage.cancelDealButton().click();
+
+      cy.url().should('eq', relative(`/case/${dealId}/cancellation/reason`));
+      cy.clickContinueButton();
+
+      cy.url().should('eq', relative(`/case/${dealId}/cancellation/bank-request-date`));
+      cy.keyboardInput(bankRequestDatePage.bankRequestDateDay(), dateConstants.todayDay);
+      cy.keyboardInput(bankRequestDatePage.bankRequestDateMonth(), dateConstants.todayMonth);
+      cy.keyboardInput(bankRequestDatePage.bankRequestDateYear(), dateConstants.todayYear);
+      cy.clickContinueButton();
+
+      cy.url().should('eq', relative(`/case/${dealId}/cancellation/effective-from-date`));
+      cy.keyboardInput(effectiveFromDatePage.effectiveFromDateDay(), dateConstants.tomorrowDay);
+      cy.keyboardInput(effectiveFromDatePage.effectiveFromDateMonth(), dateConstants.tomorrowMonth);
+      cy.keyboardInput(effectiveFromDatePage.effectiveFromDateYear(), dateConstants.tomorrowYear);
+      cy.clickContinueButton();
+    });
+
+    it('should render page correctly', () => {
+      cy.url().should('eq', relative(`/case/${dealId}/cancellation/check-details`));
+
+      backLink();
+      checkDetailsPage.reasonResponse();
+      checkDetailsPage.reasonResponse().should('have.text', '-');
+
+      checkDetailsPage.reasonLink();
+
+      checkDetailsPage.bankRequestDateResponse();
+      checkDetailsPage.bankRequestDateResponse().should('have.text', format(dateConstants.today, 'd MMMM yyyy'));
+
+      checkDetailsPage.bankRequestDateLink();
+
+      checkDetailsPage.effectiveFromResponse();
+      checkDetailsPage.effectiveFromResponse().should('have.text', format(dateConstants.tomorrow, 'd MMMM yyyy'));
```

Comment:
Can we use `cy.assertText` instead of `should('have.text'` please 🙏

Reply:
Ah yes of course!! Have changed thank you!